src/UnitTests.Shared/RunnerUtilities.cs

```diff
@@ -85,6 +87,15 @@ public static string RunProcessAndGetOutput(string process, string parameters, o
                 UseShellExecute = false,
                 Arguments = parameters
             };
+
+            if(environmentVariables != null)
```

Comment:
We've worked around the inability to set environment variables in a different way elsewhere. (See [here](https://github.com/dotnet/msbuild/blob/6945be012e19946fb89118f1c1b37c5ff33c5a9b/src/MSBuild.UnitTests/XMake_Tests.cs#L2291-L2320).) Do you think it would be reasonable to combine them?

Reply:
I imagine so.  Are you proposing a separate class, or moving the usage in XMake_Tests to use what I've done here, or something else?

Reply:
I'd do the latter or move what's in XMake_Tests here and call it. I don't have a strong preference between them, but I'd like it to be in one spot.